packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -179,14 +176,19 @@ export class CdkToolkit {
           stream,
         }) : undefined;
 
+        const diff = cfnDiff.fullDiff(currentTemplate, stack.template, changeSet);
+        if (!quiet || !diff.isEmpty) {
+          stream.write(format('Stack %s\n', chalk.bold(stack.displayName)));
+        }
+
         if (resourcesToImport) {
           stream.write('Parameters and rules created during migration do not affect resource configuration.\n');
         }
 
         const stackCount =
         options.securityOnly
-          ? (numberFromBool(printSecurityDiff(currentTemplate, stack, RequireApproval.Broadening, changeSet)) > 0 ? 1 : 0)
-          : (printStackDiff(currentTemplate, stack, strict, contextLines, quiet, changeSet, stream) > 0 ? 1 : 0);
+          ? (numberFromBool(printSecurityDiff(currentTemplate, stack, RequireApproval.Broadening, changeSet, diff)) > 0 ? 1 : 0)
+          : (printStackDiff(currentTemplate, stack, strict, contextLines, quiet, changeSet, stream, diff) > 0 ? 1 : 0);
```

Comment:
Let's not pass the `diff` in the same function that takes the inputs to create the diff (the old template and new template). Instead, let's not pass the diff into `printStackDiff`, but move the
```
          stream.write(format('Stack %s\n', chalk.bold(stack.displayName)));
```

to `printStackDiff`.